icechunk/src/cli/config.rs

```diff
@@ -0,0 +1,125 @@
+use std::{collections::HashMap, path::PathBuf, str::FromStr};
+
+use serde::{Deserialize, Serialize};
+
+use crate::config::{RepositoryConfig, S3Options, S3StaticCredentials};
+
+// Redefine to remove the Refreshable field
```

Comment:
It seems like quite a bit of work just to remove that field. Remember that you'll need also to do the same for GCP and other credential types. Can we live with `Refreshable` even if we fail on it?

Reply:
Initially I wanted to but the `Refreshable` field contains a trait object for which I couldn't simply derive `PartialEq`, preventing me from testing these configs. Having this subset-enum seemed like the simplest solution. Do you think there's a better one?

Reply:
Yeah, you lose `PartialEq` but it may be the better than maintaining another parallel hierarchy.

 We do a lot of `assert!(matches!(...))` in tests when we cannot use `PartialEq`

Reply:
Ah, nice, I didn't know about `assert!(matches!(...))`. Changed in https://github.com/earth-mover/icechunk/pull/716/commits/7cdd08768fab1796069d0ca144a4320b1dff3043